src/ffs/private/tree.ts

```diff
@@ -73,8 +83,24 @@ export class PrivateTree extends PublicTree {
     }
   }
 
+  updatePinMap(key: string, pinList: CID[]): Tree {
+    return new PrivateTree(this.links, this.version, this.key, {
+      ...this.pinMap,
+      [key]: pinList
+    })
+  }
+
   copyWithLinks(links: Links): Tree {
-    return new PrivateTree(links, this.key)
+    return new PrivateTree(links, this.version, this.key, this.pinMap)
+  }
+
+  pinList(): CID[] {
+    return Object.entries(this.pinMap).reduce((acc, cur) => {
+      const [parent, children] = cur
+      return acc
+              .concat([parent])
+              .concat(children)
```

Comment:
💅 Any reason to not use the spread operator here?

Reply:
Nope. good call :+1: